"Avoid intersections to avoid overlap" Fixes #3336 (#4223)
qgis/QGIS-Documentation: source/docs/user_manual/working_with_vector/editing_geometry_attributes.rst

   the default snapping mode, tolerance value, and units, which will populate
   the :guilabel:`Snapping Options` dialog.

.. index:: Snapping on intersections

Enable snapping on intersections
---------------------------------

Another available option is to use |snappingIntersection| :guilabel:`snapping on

> intersection`, which allows you to lock on geometry intersections of

Reviewer: ```suggestion
intersection`, which allows you to snap to geometry intersections of
```

Author: I was trying to avoid repeating too many words, but now that I think of it, lock may be too strong and inaccurate. Specially because lock is used in advanced digitising

> snapping enabled layers, even if there's no vertices in there.

Reviewer: ```suggestion
snapping enabled layers, even if there are no vertices at the intersections.
```

functionalities in the :guilabel:`Snapping options...` dialog in the
:menuselection:`Settings` (or :menuselection:`File`) menu. Here, you can
define |checkbox| :guilabel:`Enable topological editing`, and/or for polygon

> layers, activate the |checkbox| :guilabel:`Avoid overlap` option.

Reviewer 2: Can you review this paragraph, please? It refers to a dialog that not exist (actually, not triggered that way) and "Enable topological editing" is not a checkbox but a pushbutton

Author: Uhmm... probably all the other sections below also need some rewording (topological editing, snap in intersections...

Topological editing
===================

Apart from snapping options, the :guilabel:`Snapping options...`` dialog
(:menuselection:`Project --> Snapping options) and the
:guilabel:`Snapping` toolbar allow you to enable and disable some topological
functionalities.

.. There, you can
 define |checkbox| :guilabel:`Enable topological editing`, and/or for polygon

>  layers, activate the |checkbox| :guilabel:`Avoid overlap` option.

Author: Ah! That one is to remove.

Enable topological editing
--------------------------

> The |topopologicalEditing| :sup:`Topological editing` button helps to edit and

Reviewer: ```suggestion
The |topopologicalEditing| :sup:`Topological editing` button helps when editing and
```

> to maintain features with common boundaries. With this option enabled, QGIS

Reviewer: ```suggestion
maintaining features with common boundaries. With this option enabled, QGIS
```

Author: That one was a grammarly suggestion, I believe we should keep the two verbes in the same tense.

Reviewer: Agree.  I have modified the suggestion.

> 'detects' which boundaries are shared by the features; When you move common

Reviewer: ```suggestion
'detects' boundaries that are shared by the features; When you move common
```

> vertices/segments, QGIS will also move them on the neighboring features.

Reviewer: ```suggestion
vertices/segments, QGIS will also move them in the geometries of the neighboring features.
```

Topological editing works with features from different layers, as long as the
layers are visible and in editing mode.

.. index:: Avoid overlap
   seealso: Avoid overlap; Topology

Avoid overlap of new polygons
-----------------------------

When the snapping mode is set to ``Advanced configuration``, for polygon layers,
there's an option called |checkbox| :guilabel:`Avoid overlap`. This option
prevents you from drawing new features that overlap existing ones in the

> selected layer, which allows a quicker digitizing of adjacent polygons.

Reviewer: ```suggestion
selected layer, speeding up digitizing of adjacent polygons.
```

Author: Ah!! Much better!!

With avoid overlap enabled, if you already have one polygon, you can digitize

> a second one such that both intersect. QGIS will cut the second polygon to the

Reviewer: ```suggestion
a second one such that they intersect. QGIS will then cut the second polygon to the
```

The |topopologicalEditing| :sup:`Topological editing` button helps to edit and
to maintain features with common boundaries. With this option enabled, QGIS
'detects' which boundaries are shared by the features; When you move common
vertices/segments, QGIS will also move them on the neighboring features.

Topological editing works with features from different layers, as long as the
layers are visible and in editing mode.

.. index:: Avoid overlap
   seealso: Avoid overlap; Topology

Avoid overlap of new polygons
-----------------------------

When the snapping mode is set to ``Advanced configuration``, for polygon layers,
there's an option called |checkbox| :guilabel:`Avoid overlap`. This option
prevents you from drawing new features that overlap existing ones in the
selected layer, which allows a quicker digitizing of adjacent polygons.

With avoid overlap enabled, if you already have one polygon, you can digitize
a second one such that both intersect. QGIS will cut the second polygon to the
boundary of the existing one. The advantage is that you don't have to
digitize all vertices of the common boundary.

.. note:: If the new geometry is totally covered by existing ones, it gets

>    cleared, and QGIS will show an error message.

Reviewer 2: does it? Never noticed.

Author: It looks like previously in some cases the feature would stay with geometry. I wasn't able to reproduce that, and qgis shows a message as soon as you right-click to finish the geometry.

.. warning:: **Use cautiously the** :guilabel:`Avoid overlap` **option**

   Since this option will cut new overlapping geometries of any polygon layer,

>    you can get unexpected geometries if you forget to uncheck it when longer

Reviewer 2: ```suggestion
   you can get unexpected geometries if you forget to uncheck it when no longer
```
?
